src/Factory/OAuth2ServerFactory.php

```diff
@@ -88,6 +88,12 @@ private static function createStorageFromAdapter($adapter, array $config, Servic
             case 'mongo':
                 return self::createMongoAdapter($config, $services);
             default:
+                if ($services->has($adapter)) {
```

Comment:
https://github.com/zfcampus/zf-oauth2/issues/117

@weierophinney and @ezimuel pushed back strongly on the idea of pulling a service this way (line#91).  Instead of using adapters through createStorageFromAdapter storage is used: https://github.com/API-Skeletons/zf-oauth2-doctrine#configuration-with-zfcampuszf-mvc-auth

So, there's two ways to do it in zf-oauth2: 1. pdo & mongo.  2. use a service.  I tried to change 1. just as you are asking here and was directed to 2.